corehq/motech/repeaters/tasks.py

```diff
@@ -66,15 +69,36 @@ def clean_logs():
     queue=settings.CELERY_PERIODIC_QUEUE,
 )
 def check_repeaters():
+    # this creates a task for all partitions
+    # the Nth child task determines if a lock is available for the Nth partition
+    for current_partition in range(CHECK_REPEATERS_PARTITION_COUNT):
+        check_repeaters_in_partition.delay(current_partition, CHECK_REPEATERS_PARTITION_COUNT)
+
+
+def iterate_repeat_record_ids_for_partition(start, partition, total_partitions):
+    for record_id in iterate_repeat_record_ids(start, chunk_size=100000):
+        if hash(record_id) % total_partitions == partition:
+            yield record_id
+
+
+def iterate_repeat_records_for_partition(start, partition, total_partitions):
+    # chunk the fetching of documents from couch
+    for chunked_ids in chunked(iterate_repeat_record_ids_for_partition(start, partition, total_partitions), 10000):
```

Comment:
10k seems like a lot of docs to fetch at once (fine for IDs, though), though I don't have anything to back that up.  Did you also do profiling on doc lookups, or just ID lookups?